catala-lsp.opam

```diff
@@ -17,7 +17,7 @@ build: [
 depends: [
   "ocaml"     {>= "4.14.1"}
   "dune"      { >= "3.0" }
-  "catala"    { = "dev" }
+  "catala"
```

Comment:
I approve. When we release we can sync `catala-lsp` and `catala` together through here. But let's wait from @vincent-botbol when he gets better.